src/H5VLcallback.c

```diff
@@ -3572,30 +3570,10 @@ H5VL__file_open_find_connector_cb(H5PL_type_t plugin_type, const void *plugin_in
     vol_cb_args.args.is_accessible.fapl_id    = fapl_id;
     vol_cb_args.args.is_accessible.accessible = &is_accessible;
 
-    /* Store current error stack size */
-    if ((num_errors = H5Eget_num(H5E_DEFAULT)) < 0)
-        HGOTO_ERROR(H5E_ERROR, H5E_CANTGET, H5_ITER_ERROR, "can't get current error stack size");
+    if (H5VL_file_specific(NULL, &vol_cb_args, H5P_DATASET_XFER_DEFAULT, H5_REQUEST_NULL) < 0)
```

Comment:
This seems like another case where we need this to be a 'try'. The previous code was treating it like a 'try' by popping any errors that occurred from the error stack if the is_accessible call failed, but trying to keep the error messages (if any) around in the case where the file isn't accessible. Now if the is_accessible call fails, it just fails the iteration, which is not quite the behavior wanted here.

Reply:
It is effectively a 'try' already - it's returning the flag in a parameter.  So, any failure means that the flag couldn't be determined, which is a real error.   I think that a file not being accessible isn't an error, it's just not accessible.

Reply:
The original intention (for better or worse) was to ignore errors that occur during the `H5VL_file_specific` call, treat them as the file not being accessible with the current VOL connector and then move on to the next one so that each available connector is tried. IIRC, the native VOL used to return `FAIL` when a file wasn't accessible, but I believe that has been fixed so it will return `false` now. I'm not sure whether or not the expected behavior of `H5VL_file_specific` in terms of `FAIL` vs `false` is really documented though.

Reply:
I believe that it would be more consistent to have this routine act like the rest of the library's 'try' routines, where it is an actual error to not be able to answer the question "is this file accessible?".  That was the original intention of the 'is_accessible' callback.  As you said, the native VOL connector was already discriminating between actual errors and 'false', so it's behaving correctly.

Reply:
I'm not particularly stuck on the current implementation since this is functionality that is mostly unused outside of niche cases, but it would definitely be good to make sure that the distinction between `FAIL` and `false` for the is_accessible callback is documented appropriately.

Reply:
Yes, definitely, I'll add some blurbs for it in the morning.

Reply:
Done

Reply:
@jhendersonHDF - How's this look?